pydantic_extra_types/iban.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import Any
+
+from pydantic import GetCoreSchemaHandler
+from pydantic_core import PydanticCustomError, core_schema
+
+try:
+    import schwifty
+except ModuleNotFoundError:  # pragma: no cover
+    raise RuntimeError(
+        'The `iban` module requires "schwifty" to be installed. You can install it with "pip install schwifty".'
+    )
+
+
+class Iban(str):
+    def __init__(self, iban: str):
+        self.iban = self.validate_iban_digits(iban)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.general_after_validator_function(
+            cls.validate,
+            core_schema.str_schema(),
+        )
+
+    @classmethod
+    def validate(cls, __input_value: str, _: core_schema.ValidationInfo) -> str:
+        return cls(__input_value)
+
+    @classmethod
+    def validate_iban_digits(cls, iban: str) -> schwifty.iban.IBAN:
+        if not isinstance(iban, str):
+            raise PydanticCustomError('iban_digits', 'IBAN is invalid')  # pragma: no cover
+        return schwifty.IBAN(iban)
```

Comment:
cc @Kludex 

I test the implementation it worked fine also I added some tests to enhance the coverage!

I notice that this package has his own validation that's why I can't do a test for raising `PydanticCustomError`

```py
def test_validate_iban_digits():
    digits = '1234567890123456'
    assert Iban.validate_iban_digits(digits) is None
    with pytest.raises(PydanticCustomError, match='IBAN is invalid'):
        Iban.validate_iban_digits('hello')
```

its basically the only thing that's not making the coverage 100%

Reply:
I believe, `core_schema.str_schema` provides the `isinstance(iban, str)` check already. I would argue in favor of removing such a check from `validate_iban_digits` completely.